app/models/run.rb

```diff
@@ -255,6 +255,170 @@ def segments_with_groups
     @segments_with_groups
   end
 
+  # split tries to split this run by filling in the first unfilled segment with a time. If more is true and this split
+  # would cause the run to complete, a new segment will be created at the end of the run and left blank for the next
+  # split to fill. This can be used to perform blind runs where the runner doesn't know how many splits there will be.
+  # Just make sure to eventually call this with more=false in order to actually finish the run.
+  def split(more: false, realtime_end_ms:, gametime_end_ms:)
```

Comment:
This function is going to throw a bunch of rubocop warnings.

Reply:
Damn wow you were right. Just pushed a commit that fixed a lot of Rubocop stuff, but this function and the controller function are still too long,  too high AbcSize, and too high complexity. For now I put a Rubocop TODO comment on it, I'm about to get on a plane and will think about if refactoring should be done now or punted on so we can unblock other things first.